src/monitorService.ts

```diff
@@ -95,9 +95,6 @@ export class TxMonitorService {
     } catch(error) {
       this.logger.error(`Resending transaction with hash ${transaction.hash} failed, ${error.message}`);
     }
-
-    this.logger.debug(`Deleting transaction ${transaction.hash} from storage`);
-    await this.wallet.walletStorage.deleteTransaction(transaction.hash);
```

Comment:
Aren't this suppose to remove confirmed transactions?
Also If tx has error different than "underpriced" it should be deleted rather than submitting indefinitely?

Reply:
Ups, just realized this deletes the old transaction is storage that this one replaces 
This should happenn after sucessful sendTransaction and it should log error and not delete if error is thrown